app/gl_js.go

```diff
@@ -65,5 +74,8 @@ func (c *glContext) Refresh() error {
 }
 
 func (w *window) NewContext() (context, error) {
+	if w.contextStatus != contextStatusOkay {
+		return nil, gpu.ErrDeviceLost
```

Comment:
Why can NewContext return ErrDeviceLost? NewContext is only ever called as a result of a FrameEvent, which is sent by the driver. In other words, the driver should suppress FrameEvents while the context is lost.

Reply:
Removing it doesn't work, or will require more changes. Seems that new frames is generated as response from clicks or other events.

Reply:
app.Window only renders in response to FrameEvents, and os_js.go is the only source of those. Can't you check the context status before invoking requestAnimationFrame, and before sending a FrameEvent in draw? You also need to restart the animation loop when the context recovers.

Reply:
I don't think this is resolved yet; NewContext can still return ErrDeviceLost.

Reply:
That was removed. However, the code is a little bit more complex. :S